artifactory/commands/buildinfo/adddependencies.go

```diff
@@ -132,6 +168,40 @@ func (badc *BuildAddDependenciesCommand) collectDependenciesBySpec() (map[string
 	return dependenciesPaths, errorOccurred
 }
 
+func (badc *BuildAddDependenciesCommand) collectDependenciesByReader(reader *content.ContentReader) (success, fail int, err error) {
+	if badc.dryRun {
+		success, err = reader.Length()
+		return
+	}
+	count := 0
+	var buildInfoDependencies []buildinfo.Dependency
+	for resultItem := new(specutils.ResultItem); reader.NextRecord(resultItem) == nil; resultItem = new(specutils.ResultItem) {
+		buildInfoDependencies = append(buildInfoDependencies, convertSearchResultToDependencies(*resultItem))
+		count++
+		if count > clientutils.MaxBufferSize {
+			err := badc.saveDependenciesToFileSystem(buildInfoDependencies)
```

Comment:
Getting an error here seems critical enough. I don't think there's a point in attempting to continue saving the rest of the depedencies as build-info partials. I think we should retuen the error now.